src/web/components/StickyBottomBanner/PuzzlesBanner.tsx

```diff
@@ -0,0 +1,238 @@
+};
+
+type BuildPayloadProps = BaseProps & {
+	countryCode: string;
+	hasConsentedToArticleCounts: boolean;
+};
+
+type CanShowProps = BaseProps & {
+	asyncCountryCode: Promise<string>;
+	remotePuzzlesBannerConfig: boolean;
+};
+
+// TODO specify return type (need to update client to provide this first)
+const buildPayload = (props: BuildPayloadProps) => {
+	return {
+		tracking: {
+			ophanPageId: window.guardian.config.ophan.pageViewId,
+			platformId: 'GUARDIAN_WEB',
+			clientName: 'dcr',
+			referrerUrl: window.location.origin + window.location.pathname,
+		},
+		targeting: {
+			alreadyVisitedCount: props.alreadyVisitedCount,
+			shouldHideReaderRevenue: props.shouldHideReaderRevenue,
+			isPaidContent: props.isPaidContent,
+			showSupportMessaging: !shouldHideSupportMessaging(props.isSignedIn),
+			engagementBannerLastClosedAt: props.engagementBannerLastClosedAt,
+			subscriptionBannerLastClosedAt:
+				props.subscriptionBannerLastClosedAt,
+			mvtId: Number(getCookie('GU_mvt_id')),
+			countryCode: props.countryCode,
+			weeklyArticleHistory: getWeeklyArticleHistory(),
+			hasOptedOutOfArticleCount: !props.hasConsentedToArticleCounts,
+		},
+	};
+};
+
+const getPuzzlesBanner = (
+	meta: { [key: string]: any },
+	url: string,
+): Promise<Response> => {
+	const json = JSON.stringify(meta);
+	return fetch(url, {
+		method: 'post',
+		headers: { 'Content-Type': 'application/json' },
+		body: json,
+	});
+};
+
+export const canShow = async ({
+	remotePuzzlesBannerConfig,
+	isSignedIn,
+	asyncCountryCode,
+	contentType,
+	sectionName,
+	shouldHideReaderRevenue,
+	isMinuteArticle,
+	isPaidContent,
+	isSensitive,
+	tags,
+	contributionsServiceUrl,
+	alreadyVisitedCount,
+	engagementBannerLastClosedAt,
+	subscriptionBannerLastClosedAt,
+}: CanShowProps): Promise<CanShowResult> => {
+	const isPuzzlesPage =
+		window.guardian.config.page.section === 'crosswords' ||
+		window.guardian.config.page.series === 'Sudoku';
+
+	if (shouldHideReaderRevenue) {
+		// We never serve Reader Revenue banners in this case
+		return { result: false };
+	}
+
+	if (isPuzzlesPage && remotePuzzlesBannerConfig) {
+		const countryCode = await asyncCountryCode;
+		const hasConsentedToArticleCounts = await getArticleCountConsent();
+		const bannerPayload = buildPayload({
+			isSignedIn,
+			countryCode,
+			contentType,
+			sectionName,
+			shouldHideReaderRevenue,
+			isMinuteArticle,
+			isPaidContent,
+			tags,
+			contributionsServiceUrl,
+			isSensitive,
+			alreadyVisitedCount,
+			engagementBannerLastClosedAt,
+			subscriptionBannerLastClosedAt,
+			hasConsentedToArticleCounts,
+		});
+		return getPuzzlesBanner(
+			bannerPayload,
+			`${contributionsServiceUrl}/puzzles`,
+		)
+			.then(checkForErrors)
+			.then((response) => response.json())
+			.then((json: { data?: any }) => {
+				if (!json.data) {
+					return { result: false };
+				}
+
+				const { module, meta } = json.data;
+
+				return { result: true, meta: { module, meta } };
```

Comment:
This function is called `canShow` but it doesn't return a boolean. Maybe this logic can be broken down into  two steps, one function to check the logic at the start, returning a boolean, and then another which makes the fetch call?

Reply:
I agree the names aren't great, but I feel like this would be seriously out of scope- it would involve changing a lot of code unrelated to the puzzles banner.

Reply:
Fair enough. You've got a better overview on what code is being used where so I'm happy to defer to you on that